library/csv.pl

```diff
@@ -396,21 +396,32 @@
     !,
     Row = end_of_file.
 csv_read_row(Stream, Row, Record) :-
-    read_lines_to_codes(Stream, Codes),
+    read_lines_to_codes(Stream, Codes, even),
     phrase(row(Row0, Record), Codes),
     !,
     Row = Row0.
 
-read_lines_to_codes(Stream, Codes) :-
+read_lines_to_codes(Stream, Codes, Quotes) :-
     read_string(Stream, "\r\n", "\r\n", Sep, String),
     string_codes(String, Codes0),
     (   Sep == -1
     ->  Codes = Codes0
-    ;   Codes = Codes0
+    ;   check_quotes(Codes0, Quotes, even)
+    ->  Codes = Codes0
     ;   append(Codes0, [0'\n|Tail], Codes),
-        read_lines_to_codes(Stream, Tail)
+        read_lines_to_codes(Stream, Tail, odd)
     ).
 
+check_quotes([], Quotes, Quotes) :- !.
+
```

Comment:
No empty line I guess normally. Applies to the following as well.

Reply:
There is a caveat with empty lines. Inside a quoted field an empty line should be preserved, but it’s not (due to padding in `read_string/5`). Otherwise, empty line should fail `csv_read_file_row/3`. I will look into it closer...

Reply:
Ok. Here I was referring to the code itself. :) Seems like a convention at least in this file that no empty line between simple predicates.

Reply:
Got you! :) I'm externalizing my previous comment then.